server/scsynth/SC_PortAudio.cpp

```diff
@@ -293,26 +293,23 @@ int SC_PortAudioDriver::PortAudioCallback( const void *input, void *output,
 	return paContinue;
 }
 
-void SC_PortAudioDriver::GetPaDeviceFromName(const char* device, int* mInOut) {
+void SC_PortAudioDriver::GetPaDeviceFromName(const char* device, int* mInOut, bool isInput) {
 
 	const PaDeviceInfo *pdi;
 	const PaHostApiInfo *apiInfo;
 	char devString[256];
 	PaDeviceIndex numDevices = Pa_GetDeviceCount();
-	mInOut[0] = paNoDevice;
-	mInOut[1] = paNoDevice;
+	*mInOut = paNoDevice;
 
-	// This tries to find one or two devices that match the given name (substring)
-	// might cause problems for some names...
 	for( int i=0; i<numDevices; i++ ) {
 		pdi = Pa_GetDeviceInfo( i );
 		apiInfo = Pa_GetHostApiInfo(pdi->hostApi);
 		strcpy(devString, apiInfo->name);
 		strcat(devString, " : ");
 		strcat(devString, pdi->name);
 		if (strstr(devString, device)) {
-			if (pdi->maxInputChannels > 0) mInOut[0] = i;
-			if (pdi->maxOutputChannels > 0) mInOut[1] = i;
+			if (((pdi->maxInputChannels > 0) && isInput) || ((pdi->maxOutputChannels > 0) && !isInput))
```

Comment:
would probably be clearer if you split this into an if-else and put the ioType checks first